pkg/provider/consulcatalog/config.go

```diff
@@ -33,6 +34,10 @@ func (p *Provider) buildConfiguration(ctx context.Context, items []itemData) *dy
 			continue
 		}
 
+		if item.ExtraConf.ConnectEnabled {
+			confFromLabel.TLS = tlsFromLeafCert(tlsInfo)
```

Comment:
apparently this is not enough to propagate the certificates all the way up to the connection.
any help here would be greatly appreciated.